lib/ssl/src/ssl_record.erl

```diff
@@ -202,6 +203,33 @@ set_renegotiation_flag(Flag, #{current_read := CurrentRead0,
 		      pending_read => PendingRead,
 		      pending_write => PendingWrite}.
 
+%%--------------------------------------------------------------------
+-spec set_max_fragment_length(term(), connection_states()) -> connection_states().
+%%
+%% Description: Set maximum fragment length in all connection states
+%%--------------------------------------------------------------------
+set_max_fragment_length(#max_frag_enum{enum = MaxFragEnum},
+                        #{current_read := CurrentRead0,
+                          current_write := CurrentWrite0,
+                          pending_read := PendingRead0,
+                          pending_write := PendingWrite0}
+                        = ConnectionStates) ->
+    MaxFragmentLength = if MaxFragEnum == 1 -> ?MAX_FRAGMENT_LENGTH_BYTES_1;
+                           MaxFragEnum == 2 -> ?MAX_FRAGMENT_LENGTH_BYTES_2;
+                           MaxFragEnum == 3 -> ?MAX_FRAGMENT_LENGTH_BYTES_3;
+                           MaxFragEnum == 4 -> ?MAX_FRAGMENT_LENGTH_BYTES_4
```

Comment:
I'm just thinking if we should add a guard expression `true` here to properly handle a fault even at this level.